test/videojs-contrib-hls.test.js

```diff
@@ -1096,6 +1096,121 @@ QUnit.test('playlist 404 should blacklist media', function(assert) {
   assert.equal(this.player.tech_.hls.stats.bandwidth, 1e10, 'bandwidth set above');
 });
 
+QUnit.test('detect if the playlist stuck at the end', function(assert) {
+  let playlist;
+
+  this.player.src({
+    src: 'manifest/master.m3u8',
+    type: 'application/vnd.apple.mpegurl'
+  });
+  openMediaSource(this.player, this.clock);
+  this.standardXHRResponse(this.requests.shift());
+  let hls = this.player.tech_.hls;
+
+  playlist = hls.selectPlaylist();
+  // not stuck at playlist end when no seekable, even if empty buffer
+  // and positive currentTime
+  hls.masterPlaylistController_.seekable = function() {
+    return videojs.createTimeRange();
+  };
+  this.player.tech_.buffered = function() {
+    return videojs.createTimeRange();
+  };
+  this.player.tech_.setCurrentTime(170);
+  assert.ok(!hls.masterPlaylistController_.stuckAtPlaylistEnd_(playlist), 'not stuck at playlist end');
+
+  // not stuck at playlist end when no seekable, even if empty buffer
+  // and currentTime 0
+  this.player.tech_.setCurrentTime(0);
+  assert.ok(!hls.masterPlaylistController_.stuckAtPlaylistEnd_(playlist), 'not stuck at playlist end');
+
+  // not stuck at playlist end when no seekable but more buffer
+  this.player.tech_.buffered = function() {
+    return videojs.createTimeRange(0, 170);
+  };
+  assert.ok(!hls.masterPlaylistController_.stuckAtPlaylistEnd_(playlist), 'not stuck at playlist end');
+
+  // not stuck at playlist end when currentTime not at seekable end
+  // even if the buffer is empty
+  hls.masterPlaylistController_.seekable = function() {
```

Comment:
For all of these really short functions, when we don't need a specific context, it is sometimes nice to use the ES6 syntax: `hls.masterPlaylistController_.seekable = () => videojs.createTimeRange(0, 130);`